BedrockServer.cpp

```diff
@@ -680,7 +680,18 @@ void BedrockServer::worker(int threadId)
 }
 
 void BedrockServer::runCommand(unique_ptr<BedrockCommand>&& _command, bool isBlocking) {
-    // This takes ownership of the passed command. By calling the move constructor, the caller's unqiue_ptr is now empty, and so when the one here goes out of scope (i.e., this function
+    // If there's no sync node (because we're detaching/attaching), we can only queue a command for later.
+    // Also,if this command is scheduled in the future, we can't just run it, we need to enqueue it to run at that point.
+    // This functionality will go away as we remove the queues from bedrock, and so this can be removed at that time.
+    {
+        auto _syncNodeCopy = atomic_load(&_syncNode);
```

Comment:
Just to confirm my understanding, in Auth when we were using `runCommand` we were previously bypassing `bedrock.handleSocket` / `bedrock._acceptSockets` since that's only called in `postPoll` & `_beginShutdown`. 

This now ensures we always hit this logic to push the command queue whether we use `runCommand` or not?

Reply:
It's not so much that we bypass `handleSocket` or acceptSocket` from `runCommand`, it's just that calling `runCommand` did not do this check because *most* commands come through `handleSocket` and so it was getting checked there.

But that's not the case for commands created with `forwardCommand` and so those commands weren't having their execution time checked, because `runCommand` essentially assumed it was checked in `handleSocket`.

Now it's handled in `runCommand` regardless of where `runCommand` is called from.

If that's any clearer.